docs/DeveloperGuide.md

```diff
@@ -312,32 +312,196 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+* 1c. An identical phone number is detected.
+
+  * 1c1. TalentHub outputs an error message specifying the issue.
+
+    Use case resumes at step 1.
+
+**Use case: UC02 - Delete Contact after List**
+
+**MSS**
+
+1. Talent Manager [list contacts (UC06)](#)
+2. Talent Manager [delete contact (UC04)](#)
+
+   Use case ends.
+
+**Use case: UC03 - Delete Contact after List**
+
+**MSS**
+
+1. Talent Manager [find contacts (UC05)](#)
+2. Talent Manager [delete contact (UC04)](#)
+
+   Use case ends.
+
+**Use case: UC04 - Delete Contact**
+
+**MSS**
+
+1. Talent Manager requests to delete a specific person in the list
+2. TalentHub deletes the person
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The command format is incorrect.
+
+  * 1a1. TalentHub outputs a generic error message about incorrect command format.
+
+    Use case resumes at step 1.
+
+* 1b. The list is empty.
+
+  * 1b1. System shows an error message specifying the list is empty.
+
+    Use case resumes at step 1.
+
+* 1c. The given index is invalid.
+
+  * 1c1. System shows an error message.
+
+    Use case resumes at step 1.
+
+**Use case: UC05 - Find Contact**
+
+**MSS**
+
+1. User requests to find persons with with `keywords`
+2. System processes and list person with `keywords`
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The command format is incorrect.
+
+  * 1a1. TalentHub outputs a generic error message about incorrect command format.
+
+    Use case resumes at step 1.
+
+* 1b. The keyword is empty.
+
+  * 1b1. System shows an error message.
+
+    Use case resumes at step 1. 
+
+
+**Use case: UC06 - List Contact**
+
+**MSS**
+
+1. Talent Manager requests to list persons
+2. System shows a list of persons
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The command format is incorrect.
+
+  * 1a1. TalentHub outputs a generic error message about incorrect command format.
+
+    Use case resumes at step 1.
+
+
+**Use case: UC07 - Add Events**
+
+**MSS**
+
+1. Talent Manager requests to add a event for a specific contact
+2. System adds the event
+
+Use case ends.
+
+**Extensions**
+
+* 1a. The command format is incorrect.
+
+  * 1a1. TalentHub outputs a generic error message about incorrect command format.
+
+    Use case resumes at step 1.
+
+* 1b. A duplicate event is detected:
+  
+  * 1b1. System displays a message informing the Talent Manager of the duplicate event and does not add it.
+
+    Use case resumes at step 1.
+
+    
+**Use case: UC08 - List Events**
+
+**MSS**
+
+1. Talent Manager requests to list all events.
+2. System retrieves and displays all events in chronological order.
+   
+   Use case ends.
+
+**Extensions**
+
+* 1a. The command format is invalid.
+
+  * 1a1. TalentHub outputs a generic error message about incorrect command format.
+
+    Use case resumes at step 1.
+
+* 2a. If no events are scheduled.
+
+  * 2a1. System displays a message stating "No events.".
+
+    User case ends.
+
+
+**Use case: UC09 - Delete Events**
```

Comment:
Should we abstract out like about into delete after list, in case we have similar find event feature in the future?

Reply:
I think that is a good attempt. I will abstract it like `delete contact`.